File: apps/android-studio/app/src/main/kotlin/com/apexlions/film2/studio/tmdb/ImageUrls.kt
package com.apexlions.film2.studio.tmdb

/** Kotlin port of packages/tmdb-client/src/imageUrls.js. */
object TmdbImageUrls {
    private const val IMAGE_BASE = "https://image.tmdb.org/t/p"

    fun poster(path: String?, size: String = "w780"): String? = path?.let { "$IMAGE_BASE/$size$it" }
    fun backdrop(path: String?, size: String = "w1280"): String? = path?.let { "$IMAGE_BASE/$size$it" }
    fun profile(path: String?, size: String = "w300"): String? = path?.let { "$IMAGE_BASE/$size$it" }
    fun still(path: String?, size: String = "w500"): String? = path?.let { "$IMAGE_BASE/$size$it" }
}

File: packages/tmdb-client/src/imageUrls.js
const IMAGE_BASE = "https://image.tmdb.org/t/p";

export function posterUrl(path, size = "w780") {
  return path ? `${IMAGE_BASE}/${size}${path}` : "";
}

export function backdropUrl(path, size = "w1280") {
  return path ? `${IMAGE_BASE}/${size}${path}` : "";
}

export function profileUrl(path, size = "w300") {
  return path ? `${IMAGE_BASE}/${size}${path}` : "";
}

export function stillUrl(path, size = "w500") {
  return path ? `${IMAGE_BASE}/${size}${path}` : "";
}
